src/main/java/com/endava/model/MoneyTransfer.java

```diff
@@ -40,7 +43,10 @@
     @JoinColumn(name = "id_user")
     private User user;
 
-    @NotNull
+    @OneToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "id_domain")
+    private Domain domain;
+
     @OneToOne(fetch = FetchType.LAZY)
     @JoinColumn(name = "id_wherefrom")
     private Wherefrom wherefrom;
```

Comment:
Create enum, add Enum field here. Create 2 methods (ex.):
public boolean isIncome() { TransferType.INCOME.equals(this.getTransferType()); }
public boolean isExpense() { TransferType.EXPENSE.equals(this.getTransferType()); }